plugins/org.yakindu.sct.generator.c/src/org/yakindu/sct/generator/c/files/MetaSource.xtend

```diff
@@ -0,0 +1,42 @@
+package org.yakindu.sct.generator.c.files
+
+import org.yakindu.sct.generator.c.IContentTemplate
+import org.yakindu.sct.generator.c.extensions.GenmodelEntries
+import com.google.inject.Inject
+import org.yakindu.sct.model.sexec.ExecutionFlow
+import org.yakindu.sct.model.sgen.GeneratorEntry
+import org.yakindu.sct.generator.c.IGenArtifactConfigurations
+import org.yakindu.sct.generator.c.CGeneratorConstants
+import org.yakindu.sct.model.sexec.TimeEvent
+import org.yakindu.sct.generator.c.extensions.Naming
+
+class MetaSource implements IContentTemplate {
+	@Inject extension GenmodelEntries
+	@Inject extension Naming
+	
+	override content(ExecutionFlow flow, GeneratorEntry entry, extension IGenArtifactConfigurations locations) {
+		'''
+		«entry.licenseText»
+		
+		#include "«(flow.metaModule.h).relativeTo(flow.metaModule.c)»"				
+				
+		«CGeneratorConstants::STRING_TYPE» «flow.name.toFirstUpper»Meta_feature_names[] = {
+			"<nothing>",
+			«FOR feature : flow.scopes.map[it | it.declarations].flatten.reject(TimeEvent) SEPARATOR ","»
```

Comment:
This is slightly different to `SExecExtensions.eventAndVariableDefinitions`. Is this on purpose, or can we reuse the same code?

Reply:
good hint - code now uses proposed extension function